draft-pauly-add-resolver-discovery.md

```diff
@@ -89,33 +89,38 @@ they appear in all capitals, as shown here.
 
 This document defines the following terms:
 
-Companion DoH Server:
-: A DNS resolver that provides connectivity over HTTPS (DoH) that is designated as 
-equivalent to querying a given Direct Resolver.
+Direct Resolver:
+: A DNS resolver using any transport, encrypted or unencrypted, that is provisioned directly by a local router or a VPN.
 
 Designated Resolver:
-: A DNS resolver that is designated as a responsible resolver for a given domain or zone.
+: A DNS resolver that is designated as a responsible resolver for a given domain or zone. Designated resolvers use encrypted transports.
 
-Direct Resolver:
-: A DNS resolver using any transport that is provisioned directly by a local router or a VPN.
+Companion DoH Server:
+: A DNS resolver that provides connectivity over HTTPS (DoH) that is designated as 
+equivalent to querying a particular Direct Resolver.
 
 # Designated Resolvers
 
-- Based on DNSSEC-signed SVCB record
-- Based on confirmation of a trusted domain, listed in additional information
-- Use IP hints for addresses
+An encrypted DNS resolver, such as a DoH or DoT server, can be designated for use in resolving names within one or more zones. This means that clients can learn about an explicit mapping from a given domain or zone to one or more Designated Resolvers, and use that mapping to select the best resolver for a given query.
+
+Designating a resolver MUST rely on mutual agreement between the entity managing a zone (the Domain Owner) and the entity operating the resolver. These entities can be one and the same, or a Domain Owner can choose to designate a third-party resolver to handle its traffic. By presenting proof of this mutual agreement, the two entities assert to clients that interacting with the designated resolver will not expose information about names being resolved to an entity that would otherwise not be able to learn such information.
```

Comment:
```suggestion
Designating a resolver MUST rely on mutual agreement between the entity managing a zone (the Domain Owner) and the entity operating the resolver. These entities can be one and the same, or a Domain Owner can choose to designate a third-party resolver to handle its traffic. Proof of this mutual agreement asserts to clients that sending any query to the designated resolver exposes no more information than sending sending said query to the entity managing the corresponding zone.
```

(I had a hard time parsing the last sentence!)

Reply:
Incorporate, and fixed typo!